bayesian_changepoint_detection_experiment/03_changepoint_exploration_experiment.py

```diff
@@ -323,9 +332,61 @@ def examine_words_at_timepoint_range(
 fig.savefig(f"{str(figure_output_path)}/2002_changepoint.svg")
 fig.savefig(f"{str(figure_output_path)}/2002_changepoint.png", dpi=500)
 
+plot_values(change_metric_df, change_point_df, tok="2005")
+
+token_map = examine_words_at_timepoint_range(
+    word_model_cutoff_map,
+    [2002, 2003, 2004, 2005, 2006, 2007],
+    tok="2005",
+    topn=5,
+)
+token_map
+
+tokens = from_memberships(token_map.values())
+fig = plt.figure(figsize=(11, 8))
+upset = UpSet(
+    tokens,
+    with_lines=False,
+    show_counts=False,
+    sort_by="cardinality",
+    intersection_plot_elements=0,
+)
+upset.style_subsets(present="annual", facecolor="#1b9e77")
+upset.style_subsets(present="usa", facecolor="#7570b3")
+axes = UpSet.plot(upset, fig=fig)
+axes["shading"].set_xlabel("Order 2001, 2002, 2003, 2004, 2005, 2006, 2007")
+fig.savefig(f"{str(figure_output_path)}/2005_changepoint.svg")
+fig.savefig(f"{str(figure_output_path)}/2005_changepoint.png", dpi=500)
+
+plot_values(change_metric_df, change_point_df, tok="2010")
+
+token_map = examine_words_at_timepoint_range(
+    word_model_cutoff_map,
+    [2006, 2007, 2008, 2009, 2010, 2011],
+    tok="2010",
+    topn=5,
+)
+token_map
+
+tokens = from_memberships(token_map.values())
+fig = plt.figure(figsize=(11, 8))
+upset = UpSet(
+    tokens,
+    with_lines=False,
+    show_counts=False,
+    sort_by="cardinality",
+    intersection_plot_elements=0,
+)
+upset.style_subsets(present="london", facecolor="#1b9e77")
+upset.style_subsets(present="2011", facecolor="#7570b3")
+axes = UpSet.plot(upset, fig=fig)
+axes["shading"].set_xlabel("Order 2006, 2007, 2008, 2009, 2010, 2011")
+fig.savefig(f"{str(figure_output_path)}/2005_changepoint.svg")
+fig.savefig(f"{str(figure_output_path)}/2005_changepoint.png", dpi=500)
+
 # # Take home points
 
 # 1. Looks like my model is off by 1 for words such as corona virus and pandemic. This is probably due the way changepoint detection works and changes at the end of the series is hard to detect.
 # 2. Asthma has an association with type 2 diabetes. Thats really interesting as I just hadn't fathomed there was a connection.
 # 3. Years seem to undergo a shift in prevalent topics discussed to time indiciations. Need more data to be processed before I can truly say this trend is possible.
-# 4. More data to come ones the years have finally finished processing.
+# 4. Unfortunately, majority of my findings only arise from an increase in frequency. This might change if I were to use full text rather than abstracts.
```

Comment:
* I don't see the association of athsma and T2D from your plot above? Are you referring to somewhere else?
* What do you mean the findings are from an increase in frequency? So the words associated with asthma are also the more frequently used words for those pair of years?

Reply:
>I don't see the association of athsma and T2D from your plot above? Are you referring to somewhere else?

Good catch. Since I switched changepoint detection algorithms I'm not sure this discovery holds anymore. I'll remove that point.

>What do you mean the findings are from an increase in frequency? So the words associated with asthma are also the more frequently used words for those pair of years?

So during my last committee meeting one of my members asked if I could do a separate analysis where I see what changes using frequency vs what changes using cosine distance. Unfortunately, since I'm using abstracts, most of my changes were dominated by frequency (i.e. I can only detect changes if a word's frequency suddenly changes).